plugins/occt/module/vtkF3DOCCTReader.cxx

```diff
@@ -260,7 +270,10 @@ class vtkF3DOCCTReader::vtkInternals
     polydata->SetLines(linesCells);
 
 #if F3D_PLUGIN_OCCT_XCAF
-    polydata->GetCellData()->SetScalars(colors);
+    if (colors->GetSize() > 0)
+    {
+      polydata->GetCellData()->SetScalars(colors);
+    }
```

Comment:
not a big fan of a logic based on colors size. What about checking if color tools is available as soon as possible and not allocating colors at all if not needed ?

Reply:
the only initialization I can see is: https://github.com/f3d-app/f3d/blob/895480c57ae56ed8800f64591317db364aa7e29b/plugins/occt/module/vtkF3DOCCTReader.cxx#L90-L92

then insertions in 2 place, both conditional on `if (this->ColorTool)`:
https://github.com/f3d-app/f3d/blob/895480c57ae56ed8800f64591317db364aa7e29b/plugins/occt/module/vtkF3DOCCTReader.cxx#L151
https://github.com/f3d-app/f3d/blob/895480c57ae56ed8800f64591317db364aa7e29b/plugins/occt/module/vtkF3DOCCTReader.cxx#L257

Is there some costly allocation I'm missing somewhere? 

Or are we talking about refactoring the whole thing to de-tangle the colors from the other stuff happening in between so it can be guarded by a single `if (this->ColorTool)`? because that sound like a lot of work for a janky file format that nobody uses ;)

Reply:
> Is there some costly allocation I'm missing somewhere?

Nope

> Or are we talking about refactoring the whole thing to de-tangle the colors 

Yes. It is just a suggestion, Perfectly ok not to do it :)